src/Symfony/Bridge/Doctrine/PropertyInfo/DoctrineExtractor.php

```diff
@@ -117,7 +117,9 @@ public function getTypes($class, $property, array $context = [])
                         $typeOfField = $subMetadata->getTypeOfField($indexProperty);
                     }
 
-                    $collectionKeyType = $this->getPhpType($typeOfField);
+                    if (!$collectionKeyType = $this->getPhpType($typeOfField)) {
+                        return null;
```

Comment:
`new Type()` requires a built in type a its first argument. If we cannot guess it (for custom types), then we cannot extract the property types.